src/packagedcode/npm.py

```diff
@@ -531,26 +785,60 @@ def parse(cls, location, package_only=False):
                 version=version,
             )
 
-            # TODO: add resolved_package with its own deps
+            # TODO: what type of checksum is this?
             checksum = details.get('checksum')
             dependencies = details.get('dependencies') or {}
             peer_dependencies = details.get('peerDependencies') or {}
             dependencies_meta = details.get('dependenciesMeta') or {}
             # these are file references
             bin = details.get('bin') or []
 
+            deps_for_resolved_by_purl = {}
+            cls.update_dependencies_by_purl(
+                dependencies=dependencies,
+                scope="dependencies",
+                dependencies_by_purl=deps_for_resolved_by_purl,
+            )
+            cls.update_dependencies_by_purl(
+                dependencies=peer_dependencies,
+                scope="peerDependencies",
+                dependencies_by_purl=deps_for_resolved_by_purl,
+            )
+            cls.update_dependencies_by_purl(
+                dependencies=dependencies_meta,
+                scope="dependenciesMeta",
+                dependencies_by_purl=deps_for_resolved_by_purl,
+            )
+
+            dependencies_for_resolved = [
+                dep_package.to_dict()
+                for dep_package in deps_for_resolved_by_purl.values()
+            ]
+
+            resolved_package_mapping = dict(
+                datasource_id=cls.datasource_id,
+                type=cls.default_package_type,
+                primary_language=cls.default_primary_language,
+                namespace=ns,
+                name=name,
+                version=version,
+                dependencies=dependencies_for_resolved,
+                is_virtual=True,
+            )
+            resolved_package = models.PackageData.from_data(resolved_package_mapping)
             dependency = models.DependentPackage(
-                    purl=str(purl),
-                    extracted_requirement=version,
-                    is_resolved=True,
-                    # FIXME: these are NOT correct
-                    scope='dependencies',
-                    # TODO: get details  from metadata
-                    is_optional=False,
-                    is_runtime=True,
-                )
-            top_dependencies.append(dependency)
+                purl=str(purl),
+                extracted_requirement=version,
+                is_resolved=True,
+                resolved_package=resolved_package.to_dict(),
+                # FIXME: these are NOT correct
```

Comment:
I guess this may be weird, but is there something to fix at all? This is normal to default to "dependencies"